corehq/apps/data_dictionary/static/data_dictionary/js/data_dictionary.js

```diff
@@ -59,18 +68,31 @@ hqDefine("data_dictionary/js/data_dictionary", [
             self.deprecated(false);
         };
 
+        self.removePath = function () {
+            self.removeFHIRResourcePropertyPath(true);
+            // set back to original to delete the corresponding entry on save
+            self.fhirResourcePropPath(self.originalResourcePropPath);
+        };
+
+        self.restorePath = function () {
+            self.removeFHIRResourcePropertyPath(false);
+        };
+
         return self;
     };
 
-    var dataDictionaryModel = function (dataUrl, casePropertyUrl, typeChoices) {
+    var dataDictionaryModel = function (dataUrl, casePropertyUrl, typeChoices, fhirResourceTypes) {
         var self = {};
         self.caseTypes = ko.observableArray();
         self.activeCaseType = ko.observable();
+        self.fhirResourceType = ko.observable();
+        self.removefhirResourceType = ko.observable(false);
         self.newPropertyName = ko.observable();
         self.newGroupName = ko.observable();
         self.casePropertyList = ko.observableArray();
         self.showAll = ko.observable(false);
         self.availableDataTypes = typeChoices;
+        self.fhirResourceTypes = ko.observableArray(fhirResourceTypes);
```

Comment:
note to self: these don't need to be stored as observableArray since the collection won't change.